apps/heft/src/cli/HeftToolsCommandLineParser.ts

```diff
@@ -115,6 +119,66 @@ export class HeftToolsCommandLineParser extends CommandLineParser {
       registerAction: <TParameters>(options: ICustomActionOptions<TParameters>) => {
         const action: CustomAction<TParameters> = new CustomAction(options, actionOptions);
         this.addAction(action);
+      },
+      registerParameters: <TParameters, TAction extends HeftActionBase>(
+        options: ICustomParameterOptions<TParameters>
+      ) => {
```

Comment:
What happens if two different plugins try to register the same parameter name (or action)?

🤔 We could imagine a situation where two plugins were developed independently and never tested together. Renaming a parameter is a breaking change, so it might be difficult to get the plugin author to release a fix. We could imagine a Heft feature that remaps the names automatically (e.g. `--my-option` --> `--my-plugin-name-my-option`) -- or maybe it's safer to require the user to remap the name manually using a config setting.

But let's say that we won't implement that initially.  Instead, Heft will simply abort and print an error.  What does the error message look like?  Does it at least indicate:

- what is the parameter name causing the conflict?
- what are the names of the plugins that are trying to define this parameter?

Reply:
(The same considerations apply to `registerAction()`)

Reply:
For registerParameters, there is a check on line 131 that prints:
`throw new Error(`Duplicate parameter name: ${parameterName}`);`

But maybe it should print the action name too. Also, we could add a registry of the plugins that introduce the parameters, and then we could include the name of the conflicting plugins in the error message. 

Remapping parameters seems confusing. Ultimately, we should discourage use of parameter and action names that are likely to conflict. 

I think registerAction currently just overrides existing actions, but I have not tested it yet.